include/nbl/video/IDeviceMemoryAllocation.h

```diff
@@ -149,23 +188,41 @@ class IDeviceMemoryAllocation : public virtual core::IReferenceCounted
         //! Constant variant of getMappedPointer
         inline const void* getMappedPointer() const { return m_mappedPtr; }
 
+        struct SCreationParams
+        {
+            core::bitflag<E_MEMORY_ALLOCATE_FLAGS> allocateFlags = E_MEMORY_ALLOCATE_FLAGS::EMAF_NONE;
+            core::bitflag<E_MEMORY_PROPERTY_FLAGS> memoryPropertyFlags = E_MEMORY_PROPERTY_FLAGS::EMPF_NONE;
+            E_EXTERNAL_HANDLE_TYPE externalHandleType = E_EXTERNAL_HANDLE_TYPE::EHT_NONE;
+            void* externalHandle = nullptr;
+            const bool dedicated = false;
+            const size_t allocationSize;
```

Comment:
I'd reorder them and go
```cpp
allocationSize
allocateFlags
memoryPropertyFlags
dedicated
externalHandleType
externalHandle
```
this way it fits into 24 bytes, whereas currently its 32

Reply:
btw there's a lot of overlap between `IDeviceMemoryAllocation::SCreationParams` and `IDeviceMemoryAllocator::SallocationInfo` with members
```
size : 52
allocationFlags : 5
externalHandleType : 7
externalHandle
```

I'd hoist that out into a shared struct `IDeviceMemoryAllocation::SInfo` then make `SCreationParams` and the allocation `SAllocationInfo` inherit from it